### ISSUE TITLE
Duplicate tables are present with the same oid in mathesar_table

### ISSUE THREAD

**Issue by pavish** [centerofci/mathesar#559]
## Description
Duplicate tables are reflected in `mathesar_table` with the same oid, when trying to import a file. This is not regular. Happens intermittently.

<img width="812" alt="Screenshot 2021-08-15 at 8 09 57 PM" src="https://user-images.githubusercontent.com/11850603/129482617-af69fe8c-3fc4-4442-a6b5-ab4c460490f0.png">

The actual database only contains one table with that oid, as expected.

To note, whenever this issue occurs, one of the entries has a different value for `import_verified`.

This leads to the following error, when reflection happens the next time.

```
Request Method: GET
Request URL: http://localhost:8000/mathesar_tables/1/

Django Version: 3.1.7
Python Version: 3.9.6
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'django_filters',
 'django_property_filter',
 'mathesar']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/code/mathesar/views/frontend.py", line 83, in schema_home
    'common_data': get_common_data(request, database, schema)
  File "/code/mathesar/views/frontend.py", line 29, in get_common_data
    'schemas': get_schema_list(request, database),
  File "/code/mathesar/views/frontend.py", line 9, in get_schema_list
    Schema.objects.filter(database=database),
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/code/mathesar/models.py", line 25, in get_queryset
    reflection.reflect_db_objects()
  File "/code/mathesar/reflection.py", line 99, in reflect_db_objects
    reflect_tables_from_schema(schema)
  File "/code/mathesar/reflection.py", line 60, in reflect_tables_from_schema
    models.Table.current_objects.get_or_create(oid=oid, schema=schema)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 573, in get_or_create
    return self.get(**kwargs), False
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 433, in get
    raise self.model.MultipleObjectsReturned(

Exception Type: MultipleObjectsReturned at /mathesar_tables/1/
Exception Value: get() returned more than one Table -- it returned 2!
```

## Expected behavior
Duplicate table entries should not be present.

## To Reproduce
This issue happens intermittently, and I've not been able to reproduce it. It has occurred twice on my local instance.

**Comment by kgodey** [comment]
The best way to fix this would be to make `oid` a `unique` field in the model.

**Comment by pavish** [comment]
@kgodey Could we prioritize this? I'm facing this quite often, when importing CSVs, after which Mathesar becomes unusable. It doesn't block me (as I can just clear up the duplicate entries), but new contributors may face this and could be a blocker for them.

**Comment by kgodey** [comment]
@pavish I'll get to it after #578, unless @eito-fis you want to knock it out really quickly?

**Comment by pavish** [comment]
@kgodey @eito-fis 

I'm facing `Table OID is not unique` validation error, when creating a table using a data_file. This error is intermittent.

```
Environment:


Request Method: POST
Request URL: http://localhost:8000/api/v0/tables/

Django Version: 3.1.7
Python Version: 3.9.6
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'django_filters',
 'django_property_filter',
 'mathesar']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 573, in get_or_create
    return self.get(**kwargs), False
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 429, in get
    raise self.model.DoesNotExist(

During handling of the above exception (Table matching query does not exist.), another exception occurred:
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/code/mathesar/views/api.py", line 113, in create
    table = create_table_from_datafile(
  File "/code/mathesar/utils/tables.py", line 60, in create_table_from_datafile
    table = create_table_from_csv(data_file, name, schema)
  File "/code/mathesar/imports/csv.py", line 120, in create_table_from_csv
    table, _ = Table.objects.get_or_create(oid=db_table_oid, schema=schema, import_verified=False)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 576, in get_or_create
    return self._create_object_from_params(kwargs, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 610, in _create_object_from_params
    obj = self.create(**params)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 447, in create
    obj.save(force_insert=True, using=self.db)
  File "/code/mathesar/models.py", line 146, in save
    self.validate_unique()
  File "/code/mathesar/models.py", line 141, in validate_unique
    raise ValidationError("Table OID is not unique")

Exception Type: ValidationError at /api/v0/tables/
Exception Value: ['Table OID is not unique']
```

@kgodey Should we re-open this issue, or create a new one for this?

**Comment by pavish** [comment]
I assume this error is related to https://github.com/centerofci/mathesar/issues/577

**Comment by kgodey** [comment]
@pavish Please open a new issue, also put the request body that you're sending in there.
